apps/cli/src/main.ts

```diff
@@ -0,0 +1,96 @@
+import { Command, Option } from '@commander-js/extra-typings'
+import * as figlet from 'figlet'
+import fs from 'fs/promises'
+import path from 'path'
+import { exit } from 'process'
+import prompts from 'prompts'
+import { TemplateType, codeGenerator } from './code-generator'
+
+const titleText = figlet.textSync('Feature Board CLI')
+
+const program = new Command()
+    .description(`${titleText}\nA Code generator for feature board`)
+    .version('1.0.0') // how do we set this dynamicly
+    .option('-q, --quite', 'No output', false)
+
+if (!process.argv.slice(2).length) {
+    program.outputHelp()
+    exit(0)
+}
+
+// Code Gen
+const templateTypeChoices = ['dotnet-api']
+program
+    .command('code-gen')
+    .description('A Code generator for feature board')
+    .requiredOption('-p, --output-path <path>', 'Output location')
+    .addOption(
```

Comment:
We could probably guess this based on having a .csproj file, then when we do a JS project detect a package.json etc.

Reply:
Maybe, you run into issues when you have to dig though sub folder etc and then there's the .net projects that nest the front end under them.

We could guess if it's not provided maybe??

Reply:
Not doing as per conversation